configure.ac

```diff
@@ -0,0 +1,74 @@
+AC_INIT([chaosnet-bridge], 0.0)
+AC_CONFIG_AUX_DIR([build-aux])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+
+AC_CONFIG_SRCDIR([cbridge.c])
+
+AC_ARG_ENABLE([etherp],
+    AS_HELP_STRING([--enable-etherp], [Enable Chaos-over-Ether code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-etherp])])],
+    [enable_etherp=yes])
+enable_etherp_val=0
+if test "x$enable_etherp" = "xyes"; then
+    enable_etherp_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_ETHERP], [$enable_etherp_val], [])
+AM_CONDITIONAL([ENABLE_ETHERP], [test x$enable_etherp = xyes])
+
+AC_ARG_ENABLE([tls],
+    AS_HELP_STRING([--enable-tls], [Enable Chaos-over-TLS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-tls])])],
+    [enable_tls=yes])
+enable_tls_val=0
+if test "x$enable_tls" = "xyes"; then
+    enable_tls_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_TLS], [$enable_tls_val], [])
+AM_CONDITIONAL([ENABLE_TLS], [test x$enable_tls = xyes])
+
+AC_ARG_ENABLE([ip],
+    AS_HELP_STRING([--enable-ip], [Enable Chaos-over-IP code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-ip])])],
+    [enable_ip=yes])
+enable_ip_val=0
+if test "x$enable_ip" = "xyes"; then
+    enable_ip_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_IP], [$enable_ip_val], [])
+AM_CONDITIONAL([ENABLE_IP], [test x$enable_ip = xyes])
+
+# CHAOS_DNS // enable the DNS code (forwarder and internal host/addr parser)
+AC_ARG_ENABLE([dns],
+    AS_HELP_STRING([--enable-dns], [Enable DNS code @<:@default=yes@:>@]),
+    [AS_CASE(${enableval}, [yes], [], [no], [],
+        [AC_MSG_ERROR([bad value ${enableval} for --enable-dns])])],
+    [enable_dns=yes])
+enable_dns_val=0
+if test "x$enable_enable_dns" = "xyes"; then
+    enable_dns_val=1
+fi
+AC_DEFINE_UNQUOTED([CHAOS_DNS], [$enable_dns_val], [])
+AM_CONDITIONAL([ENABLE_DNS], [test x$enable_dns = xyes])
+
+#	CHAOS_DNS_SERVER
+#	CHAOS_ADDR_DOMAIN
+
+AC_PROG_CC
+AC_CHECK_HEADER([net/ethertypes.h],
+    [AC_DEFINE([HAVE_NET_ETHERTYPES_H], [1], 
+               [Define to 1 if you have <net/ethertypes.h>])])
+AC_CHECK_HEADER([net/ethernet.h], 
+     [AC_DEFINE([HAVE_NET_ETHERNET_H], [1], 
+                [Define to 1 if you have <net/ethernet.h>.])])
+AC_CHECK_HEADER([netpacket/packet.h],
+    [AC_DEFINE([HAVE_NETPACKET_PACKET_H], [1], 
+               [Define to 1 if you have <netpacket/packet.h>])])
+AC_HEADER_RESOLV
+AX_PTHREAD
+AX_CHECK_OPENSSL
```

Comment:
And openssl is of course only needed if TLS is enabled - but also libcrypto is needed in that case? (And again, I don't know what AX_CHECK_OPENSSL does.)

Reply:
As with most Autoconf macros, AX_CHECK_OPENSSL just checks if OpenSSL is available and sets some variables that can later be used.  This should follow with a "if we have OpenSSL, we should see that ENABLE_TLS is set, otherwise refuse to set ENABLE_TLS" -- then Makefile.am would do the right thing in compiling with or without TLS support.

Reply:
For `-lcrypto` it will do the right thing, if `-lcrypto` is needed on the platform:

`gcc -pthread  -g -O2   -o cbridge cbridge-cbridge.o cbridge-contacts.o cbridge-usockets.o cbridge-chudp.o cbridge-debug.o cbridge-dns.o cbridge-chtls.o cbridge-chether.o cbridge-chip.o  -lresolv -lssl -lcrypto `

Reply:
Here is a more detailed explanation of AX_CHECK_OPENSSL: https://www.gnu.org/software/autoconf-archive/ax_check_openssl.html